src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -139,56 +125,31 @@ public void removePerson(Person key) {
      * @param sg the SuperGroup to be added into AddressBook.
      */
     public void addSuperGroup(SuperGroup sg) {
-        if (!superGroups.containsKey(sg.getName())) {
-            superGroups.put(sg.getName(), sg);
-        }
+        superGroups.add(sg);
     }
 
     public void deleteSuperGroup(SuperGroup sg) {
-        superGroups.remove(sg.getName());
+        superGroups.remove(sg);
     }
 
     /**
      * Gets SuperGroup based on group name.
      */
     public SuperGroup findSuperGroup(String name) {
-        return superGroups.get(name);
-    }
-
-    /**
-     * Returns true if SuperGroup exists.
-     */
-    public boolean hasSubGroup(SubGroup subGroup) {
-        requireNonNull(subGroup);
-        return subGroups.containsKey(subGroup.toString());
-    }
-
-    /**
-     * Adds SubGroup into the AddressBook.
-     * @param subGroup the SuperGroup to be added into AddressBook.
-     */
-    public void addSubGroup(SubGroup subGroup) {
-        requireNonNull(subGroup);
-        subGroups.put(subGroup.toString(), subGroup);
-    }
-
-    public void deleteSubGroup(SubGroup subGroup) {
-        subGroups.remove(subGroup.toString());
-    }
-
-    /**
-     * Gets SuperGroup based on group name.
-     */
-    public SubGroup findSubGroup(String name) {
-        return subGroups.get(name);
+        for (SuperGroup superGroup: superGroups) {
+            if (superGroup.getName().equals(name)) {
+                return superGroup;
+            }
+        }
+        return null;
```

Comment:
As talked about in the PR thread, I think it's better to leave this abstraction to the UniqueList, and handle all IO/changes there. Since we have the Trie there, it should be easy to support this. We could even have the HashMap in the UniqueList. Plus, finding should be a task handed over to list, and maybe we can think of some higher level abstraction for this class, i.e. tasks directly related to the commands maybe?

If we decide not to do what was replied in the thread then feel free to ignore this comment.